plugins/inputs/system/memory.go

```diff
@@ -1,14 +1,21 @@
 package system
 
 import (
+	"bufio"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
+var linuxMeminfoPath = "/proc/meminfo" // so we can override it in tests
+
 type MemStats struct {
-	ps PS
+	ps           PS
+	LinuxMeminfo bool
```

Comment:
I mean, if that flag is true, you get meminfo using your implementation, if it is false, it uses ps.

I think we should always use your implementation and get rid of ps so that bool shouldn't be needed.